pmlb/update_dataset_files.py

```diff
@@ -97,6 +97,8 @@ def get_type(x, include_binary=False):
     if x.dtype=='float64':
```

Comment:
I don't think we want to check type at all. I think  we want to check:
- if > 10 values: continuous
- else: if whole number (use [float.is_integer() method](https://docs.python.org/3/library/stdtypes.html#float.is_integer)): categorical/binary, else: continuous

@lacava Am I understanding this right?

Reply:
I think if the type is float, then it should be continuous.

Reply:
I think if the type is accidentally float (like `year` in vinnie dataset, e.g., 1985.0, we should still consider whether it’s categorical).

Reply:
@trang1618 you can check that by casting to int() within a try/catch in python.

Reply:
@lacava I meant if type of a variable is float, do we automatically set it as continuous?

Reply:
i would try casting it to int. if it fails, it is continuous. else it is actually an int. then you can check the number of values.

Reply:
I found that some regression benchmark's endpoints has integer floats (I think it should be ordinal) and unique values < 10. It would become classification task if we don't check types.

https://www.openml.org/d/1027
https://www.openml.org/d/1028

Reply:
oh i see, it looks like the is_integer() method does the same thing. yes, I think your first comment is correct!

Reply:
@weixuanfu Hmm... you're right. Maybe we should lower the threshold for `target`?

Reply:
One of examples above has endpoints with number of unique value < 5.

Reply:
If we want to remove type checking then we need a hard-coded python list for regression and classification like before

Reply:
Can we keep the same coding for `target` and only apply the new coding to features?

Reply:
OK, sounds good.